scripts/superdesk/editor2/editor.ctrl.js

```diff
@@ -352,21 +353,59 @@ function SdTextEditorController(_, EMBED_PROVIDERS, $timeout, $element, editor,
         * Compute an id for the block with its content and its position.
         * Used as `track by` value, it allows the blocks to be well rendered.
         */
-        generateBlockId: function(block) {
-            function hashCode(string) {
+        generateBlockId: function(block, index) {
+            function hashCode(str) {
                 var hash = 0, i, chr, len;
-                if (string.length === 0) {
+                if (str.length === 0) {
                     return hash;
                 }
-                for (i = 0, len = string.length; i < len; i++) {
-                    chr   = string.charCodeAt(i);
+                for (i = 0, len = str.length; i < len; i++) {
+                    chr   = str.charCodeAt(i);
                     /*jshint bitwise: false */
                     hash  = ((hash << 5) - hash) + chr;
                     hash |= 0; // Convert to 32bit integer
                 }
                 return hash;
             }
-            return String(Math.abs(hashCode(block.body))) + String(vm.getBlockPosition(block));
+
+            // Here we strip the html attributes that have an empty assignment
+            var bodyToHash = block.body.replace(/=""/g, '');
+
+            if (block.body.match(/^<p><br><\/p>$/)) {
+                bodyToHash = bodyToHash;
```

Comment:
This is redundant, you can replace the entire conditional statement with:
```js
if (!block.body.match(/^<p><br><\/p>$/)) {
    bodyToHash = bodyToHash.replace(/^<p><br><\/p>/, '');
}
```
Parsing HTML can be tricky. What will happen when the `br` tag will be written differently? For example `</br>`?

Also, is it necessary to match before replacing? Replace will make no changes if it doesn't match, so I'm not sure the `if` is needed at all